### PR TITLE
Bump Apache Arrow to 2.0.0

### PR DESCRIPTION
**Pull request by yongtang**
This PR bumps Apache Arrow to 2.0.0, also bumps Apache Thrift to 0.13.0, and flatbuffers to 1.12.0.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tests/test_feather_eager.py [hunk 1]
```diff
@@ -37,9 +38,7 @@ def test_feather_format():
     }
     df = pd.DataFrame(data).sort_index(axis=1)
     with tempfile.NamedTemporaryFile(delete=False) as f:
-        df.to_feather(f)
-
-    df = pd.read_feather(f.name)
+        pa.feather.write_feather(df, f, version=1)
```
Comment (BryanCutler) [inline tests/test_feather_eager.py:41]: version 2 is the includes compression by default, is only version 1 working when reading reading back in?

**Review by BryanCutler — APPROVED** [review]
This look great, thanks for doing it @yongtang !  I've been wanting to upgrade Arrow for a while, but been too busy with other projects right now. The changes LGTM.

File: tests/test_feather_eager.py (hunk 1 above)
Comment (yongtang) [reply tests/test_feather_eager.py:41]: Thanks @BryanCutler for the review. It only works for version 1 now as version 2 changed the magic string from `FEA1` to `ARROW1` (and a few other changes in place). I am thinking of adding version 2 support in follow up PRs. This PR is just to bring arrow library to the latest version.

Comment (BryanCutler) [reply tests/test_feather_eager.py:41]: Ok great, SGTM

**Comment by chengji77** [comment]
Hi @yongtang , do you also need to bump the thrift version here? https://github.com/tensorflow/io/blob/318aab95d48598e62cf1d197307c3e12e72bfb1d/third_party/thrift.BUILD#L43

**Pull request merged**